### PR TITLE
SLCORE-668 Add textenterprise to the plugin allow list in connected mode

### PR DESCRIPTION
**Pull request by jblievremont**

### CODE REVIEW

File: plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/PluginsLoader.java [hunk 1]
```diff
@@ -79,10 +81,17 @@ public PluginsLoadResult load(Configuration configuration) {
...
     var pluginInstancesByKeys = instancesLoader.instantiatePluginClasses(nonSkippedPlugins);
 
-    return new PluginsLoadResult(new LoadedPlugins(pluginInstancesByKeys, instancesLoader, maybeDbdAllowedPlugins(configuration.enableDataflowBugDetection)),
+    return new PluginsLoadResult(new LoadedPlugins(pluginInstancesByKeys, instancesLoader, additionalAllowedPlugins(configuration)),
       pluginCheckResultByKeys);
   }
 
+  private static Set<String> additionalAllowedPlugins(Configuration configuration) {
+    var allowedPluginsIds = new HashSet<String>();
+    allowedPluginsIds.add("textenterprise");
```
Comment (thahnen) [inline plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/PluginsLoader.java:90]: Will it work when we allow the `sonar-text-enterprise` but SQ does not have it in stock? Like community or older editions?

Comment (jblievremont) [reply plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/PluginsLoader.java:90]: Yes: this is only a list used to filter which plugins are allowed among the ones downloaded from the server.

Comment (thahnen) [reply plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/PluginsLoader.java:90]: Perfect, I wasn't quite sure about this.

**Review by thahnen — APPROVED** [review]
LGTM

**Pull request merged**